projects/astc-encoder/astc_load_decompress_image_fuzzer.cc

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "astcenc_internal.h"
+#include "astcenccli_internal.h"
+#include <fuzzer/FuzzedDataProvider.h>
+
+unsigned int unpack_bytes(uint8_t a, uint8_t b, uint8_t c) {
+  return ((unsigned int)a) + ((unsigned int)b << 8) + ((unsigned int)c << 16);
+}
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+
+  if (size < 20)
+    return 0;
+
+  astc_compressed_image image_comp;
+  astcenc_error error;
+  astcenc_profile profile;
+  int out_bitness;
+  FuzzedDataProvider stream(data, size);
+
+  const astcenc_profile profiles[] = {ASTCENC_PRF_LDR, ASTCENC_PRF_LDR_SRGB,
+                                      ASTCENC_PRF_HDR,
+                                      ASTCENC_PRF_HDR_RGB_LDR_A};
+
+  int profile_type = stream.ConsumeIntegralInRange<int>(0, 3);
+  profile = profiles[profile_type];
+  out_bitness = 8 << (profile_type / 2);
+
+  // Avoid dividing by zero
+  uint8_t block_x = stream.ConsumeIntegral<uint8_t>();
+  if (!block_x)
+    block_x++;
+
+  uint8_t block_y = stream.ConsumeIntegral<uint8_t>();
+  if (!block_y)
+    block_y++;
+
+  uint8_t block_z = stream.ConsumeIntegral<uint8_t>();
+  if (!block_z)
+    block_z++;
+
+  std::vector<uint8_t> dim_x_vector = stream.ConsumeBytes<uint8_t>(3);
+  std::vector<uint8_t> dim_y_vector = stream.ConsumeBytes<uint8_t>(3);
+  std::vector<uint8_t> dim_z_vector = stream.ConsumeBytes<uint8_t>(3);
+  uint8_t *dim_x_data = dim_x_vector.data();
+  uint8_t *dim_y_data = dim_y_vector.data();
+  uint8_t *dim_z_data = dim_z_vector.data();
+
+  // Dimensions cannot be zero
+  unsigned int dim_x =
+      unpack_bytes(*dim_x_data, *(dim_x_data + 1), *(dim_x_data + 2));
+  if (!dim_x)
+    dim_x++;
+
+  unsigned int dim_y =
+      unpack_bytes(*dim_y_data, *(dim_y_data + 1), *(dim_y_data + 2));
+  if (!dim_x)
+    dim_x++;
+
+  unsigned int dim_z =
+      unpack_bytes(*dim_z_data, *(dim_z_data + 1), *(dim_z_data + 2));
+  if (!dim_x)
+    dim_x++;
+
+  unsigned int xblocks = (dim_x + block_x - 1) / block_x;
+  unsigned int yblocks = (dim_y + block_y - 1) / block_y;
+  unsigned int zblocks = (dim_z + block_z - 1) / block_z;
+  unsigned int buffer_size = xblocks * yblocks * zblocks * 16;
```

Comment:
Could you define `buffer_size` as a `size_t` instead of an `unsigned int`?  Also, why 16?  Does it make sense to use the `sizeof` something instead?  If this doesn't come from the `sizeof` something, please leave a comment explaining where 16 comes from, thanks!

Reply:
This is what the [reference file](https://github.com/ARM-software/astc-encoder/blob/master/Source/astcenccli_toplevel.cpp#L1074) did. I'll add a comment referencing the source.